Libraries/Animated/src/AnimatedImplementation.js

```diff
@@ -1199,11 +1243,9 @@ class AnimatedStyle extends AnimatedWithChildren {
     for (var key in this._style) {
       var value = this._style[key];
       if (value instanceof Animated) {
```

Comment:
Another contrived example, but imagine there is no value listener and you do the following:

- render a view with opacity = new Animated.Value(0)
- call anim.setValue(1) 
- drive a native animation from 1 to 0.5
- call setState({foo: 'bar'}); the view will be re-rendered with opacity=1

The bug will pop up when the following conditions are true:
- the value has changed on the JS side since the last render
- the value on the JS side and the native side are out of sync